src/js/actions/type.js

```diff
@@ -264,7 +265,7 @@ define(function (require, exports) {
             playObject = [playObject].concat(setOpacityPlayObjects);
         }
         var updatePromise = this.transfer(updateColor, document, layers, color, coalesce, true),
-            setColorPromise = locking.playWithLockOverride(document, layers, playObject, typeOptions);
+            setColorPromise = layerActionsUtil.playSimpleLayerActions(document, layers, playObject, true, typeOptions);
```

Comment:
The previous use of `playSimpleLayerActions` is for `applyCharacterStyle`, so it is necessary here to make `applyColor` works. I checked how `playSimpleLayerActions` works and looks like it will first select the desired layers for "Set" command, and then restore the previous select; this fix the issue when all layers are selected while apply fond color.